rpcs3/Emu/RSX/CgBinaryFragmentProgram.cpp

```diff
@@ -86,20 +84,28 @@ std::string CgBinaryDisasm::AddTexDisAsm()
 
 std::string CgBinaryDisasm::GetCondDisAsm()
 {
-	static const char f[4] = { 'x', 'y', 'z', 'w' };
+	static constexpr std::string_view f = "xyzw";
 
 	std::string swizzle, cond;
+	swizzle.reserve(5);
 	swizzle += f[src0.cond_swizzle_x];
 	swizzle += f[src0.cond_swizzle_y];
 	swizzle += f[src0.cond_swizzle_z];
 	swizzle += f[src0.cond_swizzle_w];
 
 	if (swizzle == "xxxx") swizzle = "x";
-	if (swizzle == "yyyy") swizzle = "y";
-	if (swizzle == "zzzz") swizzle = "z";
-	if (swizzle == "wwww") swizzle = "w";
+	else if (swizzle == "yyyy") swizzle = "y";
+	else if (swizzle == "zzzz") swizzle = "z";
+	else if (swizzle == "wwww") swizzle = "w";
 
-	swizzle = swizzle == "xyzw" ? "" : "." + swizzle;
+	if (swizzle != "xyzw"sv)
```

Comment:
```suggestion
	if (!swizzle.starts_with("xyzw"sv))
```
It was bugged before.

Reply:
Can you explain why this change is needed?
In previous version there was a strict check everywhere. And there can't be more than 4 chars, so starts_with will introduce confuse.

Reply:
Not to introduce confuse, I might suggest using .compare. But still a question.

Reply:
strncmp is more similar to .starts_with than .compare as it compares only a part of the string and not all of it.

Reply:
there can be more than 4 chars afaik.

Reply:
err, wrong line, I meant line 59.